### DISCUSSION TITLE
Feature Request: Add `wordWrap` Property to `Text`

### DISCUSSION THREAD

**Discussion by ambmcmdmem626** [excaliburjs/Excalibur#3178] [Ideas]
## Premise

When using the `maxWidth` property of `Text`, the text may wrap in the middle of a word.

<img width="200" alt="image" src="https://github.com/user-attachments/assets/3a8c1ccd-2717-4852-b023-d5f9ef85b1f4">

```
const text = new Text({
  text: 'Apple Banana Chocolate',
  maxWidth: 100,
});
```

## Main Request

I propose adding a `wordWrap` property to prevent words from wrapping in the middle. 
What are your thoughts on this?

> Based on https://github.com/excaliburjs/Excalibur/commit/51ea101c9305ae90b6552266dc474dfde9ef58da, it seems that a lot of changes might be needed...
> Also, I can manage this by creating a wrapping method on the application side, so it’s not a high priority at all!

<img width="184" alt="image" src="https://github.com/user-attachments/assets/da27f55e-2944-4d39-9b28-34014bc6c69b">

```
const text = new Text({
  text: 'Apple Banana Chocolate',
  maxWidth: 100,
  wordWrap: true
});
```

**Comment by eonarheim** [comment]
@ambmcmdmem626 Totally agree, this makes a lot of sense to me. This would be a very useful feature, I've made an issue from this discussion! https://github.com/excaliburjs/Excalibur/issues/3181
